src/CommunityToolkit.Mvvm.SourceGenerators/Extensions/SymbolInfoExtensions.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace CommunityToolkit.Mvvm.SourceGenerators.Extensions;
+
+/// <summary>
+/// Extension methods for the <see cref="SymbolInfo"/> type.
+/// </summary>
+internal static class SymbolInfoExtensions
+{
+    /// <summary>
+    /// Tries to get the resolved attribute type symbol from a given <see cref="SymbolInfo"/> value.
+    /// </summary>
+    /// <param name="symbolInfo">The <see cref="SymbolInfo"/> value to check.</param>
+    /// <param name="typeSymbol">The resulting attribute type symbol, if correctly resolved.</param>
+    /// <returns>Whether <paramref name="symbolInfo"/> is resolved to a symbol.</returns>
+    /// <remarks>
+    /// This can be used to ensure users haven't eg. spelled names incorrecty or missed a using directive. Normally, code would just
+    /// not compile if that was the case, but that doesn't apply for attributes using invalid targets. In that case, Roslyn will ignore
+    /// any errors, meaning the generator has to validate the type symbols are correctly resolved on its own.
+    /// </remarks>
+    public static bool TryGetAttributeTypeSymbol(this SymbolInfo symbolInfo, [NotNullWhen(true)] out INamedTypeSymbol? typeSymbol)
+    {
+        if ((symbolInfo.Symbol ?? symbolInfo.CandidateSymbols.SingleOrDefault()) is not ISymbol attributeSymbol ||
```

Comment:
I'm worried about the possibility for an exception from `SingleOrDefault`? But I see this code existing prior to the PR

Reply:
Oh right, that still throws if there's more than 1 item 🤦‍♂️
I'll change that, good catch ahah